File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        Stack<String> stack = new Stack<>();
        for (int i = 0; i <s.length() ; i++) {
            if(s.charAt(i)=='(')
                stack.push("(");
            else if(s.charAt(i)==')'&&stack.isEmpty()==true)
                return false;
            else
                stack.pop();

        }

        if(stack.isEmpty()==true) return true;
        else return false;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    st = []
    for i in s:
        if i=="(":
            st.append(i)
        elif i==")" and len(st) == 0:
            return False
        else:
            st.pop()
    
    if len(st) > 0:
        return False
    else:
        return True